cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -341,10 +340,12 @@ public void startStopAdapter(HttpRequest request, HttpResponder responder,
         return;
       }
       responder.sendStatus(HttpResponseStatus.OK);
-    } catch (AdapterNotFoundException e) {
+    } catch (NotFoundException e) {
       responder.sendString(HttpResponseStatus.NOT_FOUND, e.getMessage());
     } catch (InvalidAdapterOperationException e) {
       responder.sendString(HttpResponseStatus.CONFLICT, e.getMessage());
+    } catch (SchedulerException e) {
```

Comment:
we have a conention not to include the exception in HTTP responses for internal errors - because they reveal internal implementation details. 

but we should always log internal errors. For not found, that is a user error and kind of expected to happen. Logging it as an error would be very verbose so we would at most log it at debug level. But internal errors are no expected to happen and should be logged as errors, along with the context in which they happen (the app name, for example, or the URL of the request, which gives a lot of context).

I know there is a lot of existing code that does not do this right. Let's do it right here.